vpr/src/base/vpr_constraints.h

```diff
@@ -70,6 +89,12 @@ class VprConstraints {
      * Store all partitions
      */
     vtr::vector<PartitionId, Partition> partitions;
+
+    /**
+     * Store number of partitions - the number of partitions in the partitions vector.
+     * Used when echoing VprConstraints objects.
+     */
+    int num_partitions;
```

Comment:
check if you need this or you can just use partitions.size()

Reply:
get_num_partitions can just directly return partitions.size() in its implementation